docs/libcudacxx/extended_api/work_stealing.rst

```diff
@@ -0,0 +1,105 @@
+.. _libcudacxx-extended-api-work-stealing:
+
+Work stealing
+=============
+
+.. code:: cuda
+
+   namespace cuda::experimental {
+
+       template <int ThreadBlockRank = 3, typename UnaryFunction = ..unspecified..>
+           requires std::is_invocable_r_v<void, UnaryFunction, dim3>
+       __device__ void for_each_cancelled_block(UnaryFunction uf);
+
+   } // namespace cuda::experimental
+
+**WARNING**: This is an **Experimental API**.
+
+**WARNING**: This API requires C++20 or newer.
+
+This API is useful to implement work-stealing at thread-block level granularity.
+On devices with compute capability 10.0 or higher, it may leverage hardware acceleration for work-stealing.
+When compared against alternative work distribution techniques like `grid-stride loops <https://developer.nvidia.com/blog/cuda-pro-tip-write-flexible-kernels-grid-stride-loops/>`__, which distribute load statically, or against other dynamic work distribution techniques using global memory concurrency, the main advantages of this API over these alternatives are:
+
+   - It performs work-stealing dynamically: thread blocks that finish work sooner may do more work than thread blocks whose work takes longer.
+   - It may cooperate with the GPU work-scheduler to respect work priorities and perform load-balancing.
+   - It may have lower work-stealing latency than global memory atomics.
+
+For better performance, extract the thread block prologue - i.e. code and data that is common to all thread blocks (e.g. initialization, __shared__ memory allocations, common constants loaded to shared memory, etc.) - and thread block epilogue (e.g. writing back shared memory to global memory) outside the lambda passed to this API (see example below).
+
+**Mandates**:
+
+   - ``ThreadBlockRank`` equals the rank of the thread block: ``1``, ``2``, or ``3`` for one-dimensional, two-dimensional, and three-dimensional thread blocks, respectively.
+
+**Preconditions**:
+
+   - All threads of current thread-block call ``for_each_cancelled_block`` **exactly once**.
+
+**Effects**:
+
+   - Invokes ``uf`` with ``dim3 == blockIdx``, then repeatedly attempts to cancel the launch of a current grid thread block, and:
+
+      - on success, calls ``uf`` with that thread blocks ``blockIdx`` and repeats,
+      - otherwise it failed to cancel the launch of a thread block and it returns.
```

Comment:
```suggestion
      - or, if it failed to cancel the launch of another thread block, returns.
```

Reply:
I've slightly tweaked it and incorporated the suggestions.
Leaving this open cause I have not incorporated the leading "or, .." change.
The "otherwise" applies to "on success", so maybe a larger rephrasing is required.

Reply:
New wording is better. LGTM!